src/main/kotlin/org/rust/ide/annotator/RustHighlightingAnnotator.kt

```diff
@@ -31,6 +35,15 @@ class RustHighlightingAnnotator : Annotator {
 
         override fun visitTypeParam(o: RustTypeParamElement) {
             holder.highlight(o, RustColor.TYPE_PARAMETER)
+            o.typeParamBounds?.let {
+                holder.highlight(it.colon, RustColor.KEYWORD) // feels off
+                it.polyboundList.forEach {
+                    it.bound.traitRef?.let {
+                        // Should this visit path instead?
+                        holder.highlight(it.path.identifier, RustColor.TRAIT)
+                    }
+                }
+            }
```

Comment:
Do I understand correctly that you are colouring whole type parameter and then fixing particular parts? 

Why not just then:
```kotlin
override fun visitTypeParam(o: RustTypeParamElement) {
    holder.highlight(o.identifier, RustColor.TYPE_PARAMETER)
    o.typeParamBounds?.polyboundList?.forEach { polybound ->
        polybound.bound.traitRef?.let {
            holder.highlight(it.path.identifier, RustColor.TRAIT)
        }
    }
}
```